thingsboard_gateway/extensions/tcpserver/tcpserver_connector.py

```diff
@@ -0,0 +1,243 @@
+#     Copyright 2020. ThingsBoard
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+
+
+import threading
+from thingsboard_gateway.tb_utility.tb_utility import TBUtility
+from socketserver import ThreadingTCPServer, BaseRequestHandler
+import binascii
+import time
+from thingsboard_gateway.connectors.connector import Connector, log
+from thingsboard_gateway.extensions.tcpserver.tcpserver_converter import BytesTcpUplinkConverter
+
+
+global TCPServer_gateway
+global TCPServer_devices
+RequestHandlerDict = {}
+
+
+class MyBaseRequestHandler(BaseRequestHandler):
+    def setup(self):
+        self.__devices = TCPServer_devices
+        print("self.client_address:", self.client_address)
+
+    def get_name(self):
+        return 'TCP Server Connector'
+
+    def handle(self):
+        self.addr = self.request.getpeername()
+        message = "IP " + self.addr[0] + ":" + str(self.addr[1]) + " Connected..."
+        print(message)
+        device_config = {}
+        device_name = self.request.recv(128).decode()
+        print(f'receive from {self.client_address}:{device_name}')
+        RequestHandlerDict[device_name] = self.request
+
+        if self.__devices.keys().__contains__(device_name):
+            device_config = self.__devices[device_name]
+        while True:
+            time.sleep(0.1)
+            self.process_device(device_config)
+
+    def process_device(self, device_config):
+        if not device_config:
+            return
+        current_time = time.time()
+        device_responses = {"timeseries": {},
+                            "attributes": {},
+                            }
+        for config_data in device_responses:
+            if device_config["config"].get(config_data) is not None:
+                if device_config["next_" + config_data + "_check"] < current_time:
+                    send_data = device_config['config']['getDataCommand']
+                    self.request.sendall(binascii.a2b_hex(send_data))
+                    input_data = self.request.recv(128)
```

Comment:
Please move size of the data to receive into the configuration.

Reply:
ok.